stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+import json
+import requests_mock
+from enum import Enum
+from schematics.models import Model
+from schematics.types import StringType
+from schematics.exceptions import DataError
+from requests import Session, Request
+from requests.auth import HTTPBasicAuth
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+
+"""
+    Authentication Models
+"""
+
+
+class HTTPBasicAuthentication(Model):
+    username = StringType(required=True)
+    password = StringType(required=True)
+
+
+"""
+    Authentication ENUM
+"""
+
+
+class HTTPAuthenticationType(Enum):
+    BasicAuth = HTTPBasicAuthentication
+
+
+"""
+    Possible Request and Response objects
+"""
+
+
+class HTTPMethod(Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+
+class HTTPRequestType(Enum):
+    JSON = dict
+
+
+class HTTPResponseType(Enum):
+    PLAIN = str
+    JSON = dict
+
+
+"""
+    The HTTP Helper Common class is used for common functionality split between all the other
+    HTTP Helper classes.
+    Functionality:
+        - Mapped enums to lists
+        - Log messages
+        - Create a dictionary from a string
+"""
+
+
+class HTTPHelperCommon:
```

Comment:
This class is a weird mix of stuff.

Methods that are static can only just be functions.
Parameters that are constant can also just be module scoped variables.

I would rename this thing to printer, because that's the only real purpose it serves.

Reply:
Scrap that. We should use a real logger.

Reply:
I'll do some investigation to see if there is a existing logger being used within the other services

Reply:
Got removed with the updated python logging lib implementation